File: src/main/java/gg/mineral/practice/commands/config/PartiesCommand.java
package gg.mineral.practice.commands.config;

import gg.mineral.practice.commands.PlayerCommand;
import gg.mineral.practice.managers.PartyManager;
import gg.mineral.practice.util.messages.impl.ChatMessages;
import gg.mineral.practice.util.messages.impl.ErrorMessages;
import gg.mineral.practice.util.messages.impl.UsageMessages;

public class PartiesCommand extends PlayerCommand {

	public PartiesCommand() {
		super("parties", "practice.config");
	}

	@Override
	public void execute(org.bukkit.entity.Player pl, String[] args) {

		String arg = args.length > 0 ? args[0] : "";
		String toggled;

		switch (arg.toLowerCase()) {
			default:
				ChatMessages.PARTIES_COMMANDS.send(pl);
				ChatMessages.PARTIES_ENABLE.send(pl);
				ChatMessages.PARTIES_DISPLAY.send(pl);
				ChatMessages.PARTIES_SLOT.send(pl);
				return;
			case "enable":
				if (args.length < 2) {
					UsageMessages.PARTIES_ENABLE.send(pl);
					return;
				}

				toggled = args[1].toLowerCase();

				switch (toggled) {
					case "false":
						PartyManager.setEnabled(false);
						break;
					case "true":
						PartyManager.setEnabled(true);
						break;
					default:
						UsageMessages.PARTIES_ENABLE.send(pl);
						return;
				}

				ChatMessages.PARTIES_ENABLED.clone().replace("%toggled%", toggled).send(pl);
				return;
			case "setdisplay":
				if (args.length < 1) {
					UsageMessages.PARTIES_DISPLAY.send(pl);
					return;
				}

				PartyManager.setDisplayItem(pl.getItemInHand());

				if (args.length > 2) {
					PartyManager.setDisplayName(args[1].replace("&", "§"));
				}

				ChatMessages.PARTIES_DISPLAY_SET.send(pl);
				return;
			case "slot":
				if (args.length < 2) {
					UsageMessages.PARTIES_SLOT.send(pl);
					return;
				}

				int slot;
				String strSlot = args[1];
				try {
					slot = Integer.parseInt(strSlot);
				} catch (Exception e) {
					ErrorMessages.INVALID_SLOT.send(pl);
					return;
				}

				PartyManager.setSlot(slot);

				ChatMessages.PARTIES_SLOT_SET.clone().replace("%slot%", strSlot).send(pl);

				return;
		}
	}
}

File: src/main/kotlin/gg/mineral/practice/commands/config/PartiesCommand.kt
package gg.mineral.practice.commands.config

import dev.rollczi.litecommands.annotations.argument.Arg
import dev.rollczi.litecommands.annotations.command.Command
import dev.rollczi.litecommands.annotations.context.Context
import dev.rollczi.litecommands.annotations.execute.Execute
import dev.rollczi.litecommands.annotations.permission.Permission
import gg.mineral.practice.entity.appender.CommandSenderAppender
import gg.mineral.practice.managers.PartyManager
import gg.mineral.practice.managers.PartyManager.displayItem
import gg.mineral.practice.managers.PartyManager.enabled
import gg.mineral.practice.util.messages.impl.ChatMessages
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import java.util.*

@Command(name = "parties")
@Permission("practice.config")
class PartiesCommand : CommandSenderAppender {

    @Execute(name = "enable")
    fun executeEnable(@Context sender: CommandSender, @Arg toggled: Boolean) {
        enabled = toggled
        sender.send(ChatMessages.PARTIES_ENABLED.clone().replace("%toggled%", toggled.toString()))
    }

    @Execute(name = "setdisplay", aliases = ["display"])
    fun executeSetDisplay(@Context player: Player, @Arg displayName: Optional<String>) {
        displayName.ifPresent { PartyManager.displayName = it.replace("&", "§") }
        displayItem = player.itemInHand
        player.send(ChatMessages.PARTIES_DISPLAY_SET)
    }

    @Execute(name = "slot")
    fun executeSlot(@Context sender: CommandSender, @Arg slot: Int) {
        PartyManager.slot = slot
        sender.send(ChatMessages.PARTIES_SLOT_SET.clone().replace("%slot%", slot.toString()))
    }

    @Execute(aliases = ["help"])
    fun executeHelp(@Context sender: CommandSender) {
        sender.send(
            ChatMessages.PARTIES_COMMANDS,
            ChatMessages.PARTIES_ENABLE,
            ChatMessages.PARTIES_DISPLAY,
            ChatMessages.PARTIES_SLOT
        )
    }
}
